File: app/src/main/java/com/bebediary/calendar/decorators/SaturdayDecorator.java
package com.bebediary.calendar.decorators;

import android.content.Context;
import androidx.fragment.app.FragmentActivity;
import android.text.style.ForegroundColorSpan;
import com.bebediary.R;
import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;

import java.util.Calendar;

/**
 * Highlight Saturdays and Sundays with a background
 */
public class SaturdayDecorator implements DayViewDecorator {

    private final Calendar calendar = Calendar.getInstance();

    private Context context;
    public SaturdayDecorator(FragmentActivity activity) {
        context = activity;
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        day.copyTo(calendar);
        int weekDay = calendar.get(Calendar.DAY_OF_WEEK);
        return weekDay == Calendar.SATURDAY;
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.addSpan(new ForegroundColorSpan(context.getResources().getColor(R.color.mainColor)));
    }
}

File: app/src/main/java/com/bebediary/calendar/decorator/SaturdayDecorator.kt
package com.bebediary.calendar.decorator

import android.content.Context
import android.text.style.ForegroundColorSpan
import androidx.core.content.res.ResourcesCompat
import com.bebediary.R
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade
import java.util.*

/**
 * 토요일 표시 해주는 데코레이터
 */
class SaturdayDecorator(private val context: Context) : DayViewDecorator {

    private val calendar = Calendar.getInstance()
    private val textColor by lazy { ResourcesCompat.getColor(context.resources, R.color.mainColor, context.theme) }

    override fun shouldDecorate(day: CalendarDay): Boolean {
        day.copyTo(calendar)
        val weekDay = calendar.get(Calendar.DAY_OF_WEEK)
        return weekDay == Calendar.SATURDAY
    }

    override fun decorate(view: DayViewFacade) {
        view.addSpan(ForegroundColorSpan(textColor))
    }
}
